extensions/SemanticScribunto/src/LuaAskResultProcessor.php

```diff
@@ -0,0 +1,215 @@
+
+		$result = null;
+
+		if ( $this->queryResult->getCount() ) {
+
+			$result = [];
+
+			while ( $resultRow = $this->queryResult->getNext() ) {
+				$result[] = $this->getDataFromQueryResultRow( $resultRow );
+			}
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Extracts the data of a single result row in the {@see $queryResult}
+	 * and returns it as a table usable in lua context
+	 *
+	 * @param array $resultRow result row as an array of {@see ResultArray} objects
+	 *
+	 * @uses getDataFromResultArray
+	 *
+	 * @return array
+	 */
+	public function getDataFromQueryResultRow( array $resultRow ) {
+
+		$rowData = [];
+
+		/** @var ResultArray $resultArray */
+		foreach ( $resultRow as $resultArray ) {
+
+			// get key and data
+			list ( $key, $data ) = $this->getDataFromResultArray( $resultArray );
+
+			$rowData[$key] = $data;
+		}
+
+		return $rowData;
+	}
+
+	/**
+	 * Extracts the data of a single printRequest / query field
+	 *
+	 * @param ResultArray $resultArray
+	 *
+	 * @uses getKeyFromPrintRequest, getValueFromDataValue, extractLuaDataFromDVData
+	 *
+	 * @return array array ( int|string, array )
+	 */
+	public function getDataFromResultArray( ResultArray $resultArray ) {
+
+		// first, extract the key (label), if any
+		$key = $this->getKeyFromPrintRequest( $resultArray->getPrintRequest() );
+
+		$resultArrayData = [];
+
+		// then get all data that is stored in this printRequest / query field
+		/** @var DataValue $dataValue */
+		while ( ( $dataValue = $resultArray->getNextDataValue() ) !== false ) {
+
+			$resultArrayData[] = $this->getValueFromDataValue( $dataValue );
+		}
+
+		$resultArrayData = $this->extractLuaDataFromDVData( $resultArrayData );
+
+		return [ $key, $resultArrayData ];
+	}
+
+	/**
+	 * Takes an smw query print request and tries to retrieve the label
+	 * falls back to {@see getNumericIndex} if none found
+	 *
+	 * @param PrintRequest $printRequest
+	 *
+	 * @uses getNumericIndex
+	 *
+	 * @return int|string
+	 */
+	public function getKeyFromPrintRequest( PrintRequest $printRequest ) {
+
+		if ( $printRequest->getLabel() !== '' ) {
+			return $printRequest->getText( SMW_OUTPUT_WIKI );
+		}
+
+		return $this->getNumericIndex();
+	}
+
+	/**
+	 * Extracts the data of a single value of the current printRequest / query field
+	 * The value is formatted according to the type of the property
+	 *
+	 * @param DataValue $dataValue
+	 *
+	 * @return mixed
+	 */
+	public function getValueFromDataValue( DataValue $dataValue ) {
+
+		switch ( $dataValue->getTypeID() ) {
+			case '_boo':
+				// boolean value found, convert it
+				$value = in_array( strtolower( $dataValue->getShortText( SMW_OUTPUT_WIKI ) ), $this->msgTrue );
+				break;
+			case '_num':
+				// number value found
+				/** @var \SMWNumberValue $dataValue */
+				$value = $dataValue->getNumber();
+				$value = ( $value == (int)$value ) ? intval( $value ) : $value;
+				break;
+			default:
+				# FIXME ignores parameter link=none|subject
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)